client/src/research/ResearcherDataPage.js

```diff
@@ -281,156 +197,6 @@ class Analysis extends Component {
-            }))}
-          </tbody>
-        </table>
-        {this.renderGenericTable(statsForSessions)}
-      </div>
-    );
-  }
-
-  renderStatsRow(text, valueText) {
-    return (
-      <tr key={text}>
-        <td style={styles.cell}>{text}</td>
-        <td style={styles.cell}>{valueText}</td>
-      </tr>
-    );
-  }
-
-  renderGenericTable(rows) {
-    const fields = Object.keys(rows[0] || {});
-    return (
-      <table style={styles.table}>
-        <thead>
-          <tr>{fields.map((field) => <th key={field} style={styles.cell}>{field}</th>)}</tr>
-        </thead>
-        <tbody>
-          {rows.map((row) => {
-            return <tr key={JSON.stringify(row)}>
-              {fields.map((field) => {
-                return <td key={field} style={styles.cell}>{row[field]}</td>;
-              })}
-            </tr>;
-          })}
-        </tbody>
-      </table>
-    );
-  }
-
-  renderHmtcaTable(json) {
-    const csvRows = json.evidence.rows.map((row, i) => {
-      return {
-        id: row.id,
-        timestampText: moment(row.timestamp).format('MM/DD/YY  h:mm:ssa'),
-        cohortKey:row.json.cohortKey,
-        identifier: row.json.identifier,
-        sessionIdSlice: row.json.sessionId.slice(0, 6),
-        backgroundColor: hashInto(hmtcaRowKey(row), colorNames),
-        questionText: row.json.question.text,
-        responseText: row.json.responseText,
-        raw: row
-      };
-    });
-    const csvKeys = Object.keys(_.first(csvRows));
-    const identifierCount = _.uniq(json.evidence.rows.map(row => row.json.identifier)).length;
-
-    return (
-      <div>
-        <div>Identifier count: {identifierCount}</div>
-        <div><button onClick={this.doExport.bind(this, csvRows, csvKeys)}>Export table as CSV</button></div>
-
-        <table style={styles.table}>
-          <tbody>
-            {csvRows.map((csvRow, i) => {
-              return (
-                <tr key={csvRow.id}>
-                  <td style={styles.cell} title={JSON.stringify(csvRow.raw, null, 2)}>{csvRows.timestampText}</td>
-                  <td style={styles.cell}>{csvRow.cohortKey}</td>
-                  <td style={styles.cell}>{csvRow.identifier}</td>
-                  <td style={styles.cell}>{csvRow.sessionIdSlice}</td>
-                  <td
-                    style={{backgroundColor: csvRow.backgroundColor}}
-                    title={csvRow.identifier}>
-                    {csvRow.identifier}
-                  </td>
-                  <td style={styles.cell}>{csvRow.questionText}</td>
-                  <td style={styles.cell}>{csvRow.responseText}</td>
-                </tr>
-              );
-            })}
-          </tbody>
-        </table>
-      </div>
-    );
-  }
-
-  // For looking at raw events and sessions.  Hover to reveal more sensitive data.
-  renderEventsTable(json) {
-    const {analysisKey} = this.props;
-    if (analysisKey === 'HMTCA') return this.renderHmtcaTable(json);
```

Comment:
This looks like it's simplifying, but also that it's removing some code that is used to analyze other data besides the scenarios you've been focused on.  Can you say more about how removing this code impacts viewing data for other sessions or scenarios?

If removing this would break those, let's leave this code and add a branching to use the new awesome view where it works and for older analyses use this old code.  Or if it didn't really work before this change to look at the older analyses, then I think it's fine to remove this and just add a comment linking back to an older commit to point future folks in a helpful direction.

Reply:
So those stopped working once I stopped using the renderEventsTable() function and switched to using react virtualized. Are you saying to leave a comment in the actual code about this?

Reply:
Yep!  I think it's fine to break these older scenarios to make the awesome improvements you've made.  I'd suggest just adding comment that says something like "for analysis of other scenarios, look at (this URL in the git history)" as a helpful pointer for future folks.